### ISSUE TITLE
Configure Travis CI to build and test the Rust version

### ISSUE THREAD

**Issue by maximg** [maximg/thrift#3]

**Comment by gsingh93** [comment]
Thrift seems to have its own `.travis.yml` file. If we just enable the builds in travis, should that cover it? Or if you want to have a simpler travis file for just the Rust lib, why not use the one already in `lib/rs`?

**Comment by maximg** [comment]
Indeed, this issue is about adding Rust to Thrift Travis file; this should make sure that the library, the tutorial and the tests are built and run.

From my brief look at this long time ago I had a feeling that cross-language builds are not very well supported on Travis; maybe things have changed since then.

**Comment by gsingh93** [comment]
I don't have access to enable them on Travis, could you enable them? Then I can get them working whenever I have time.

**Comment by maximg** [comment]
Added to Travis, let me know if you need any help here. I will check if it is possible to give you admin rights on that Travis project.

**Comment by gsingh93** [comment]
So initially many of the builds were failing: https://travis-ci.org/maximg/thrift/builds/66051213. But after adding `sudo: required` to the travis file, the builds are working: https://travis-ci.org/maximg/thrift/builds/66102746

This makes sense, because some of the scripts require sudo. However, the original thrift file didn't have this line, and it's builds are working (or at least aren't failing because of sudo access): https://travis-ci.org/apache/thrift

Any idea why?

**Comment by maximg** [comment]
Sorry it took long time to get back on this one, I have not looked into Thrift Travis setup except for building the Rust library only in the early days. The best will be to check with Thrift folks if their Travis setup has anything special configured.

As we are not touching any C++ code we could focus on adding Rust tests for now, and check the failures in other parts of the project at a later stage.
